### PR TITLE
Fix flaky test QuicHttpIntegrationTest::DeferredLoggingWithRetransmission

### LINKED ISSUE #27841: flake in DeferredLoggingWithRetransmission
https://dev.azure.com/cncf/envoy/_build/results?buildId=139480&view=logs&j=4e2e4662-2d21-5726-b4bc-9ff5cd6a179b&t=da7afaec-274f-5dd3-65ad-8c8f961558bc&l=429

  0x1429ac0: Envoy::Quic::QuicHttpIntegrationTest_DeferredLoggingWithRetransmission_Test::TestBody()
  0xa406666: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0xa3ac4ff: testing::Test::Run()
  0xa3b05f8: testing::TestInfo::Run()
... Google Test internal frames ...

test/integration/quic_http_integration_test.cc:1415: Failure
Expected: (std::stoi(metrics.at(1))) > (0), actual: 0 vs 0
Stack trace:
  0x1428857: Envoy::Quic::QuicHttpIntegrationTest_DeferredLoggingWithRetransmission_Test::TestBody()
  0x1429ac0: Envoy::Quic::QuicHttpIntegrationTest_DeferredLoggingWithRetransmission_Test::TestBody()
  0xa406666: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0xa3ac4ff: testing::Test::Run()
  0xa3b05f8: testing::TestInfo::Run()
... Google Test internal frames ...

on msan

### PR DESCRIPTION
**Pull request by pksohn**
Fixes #27841:

- Increase the sleep from 500->1000 ms.
- Properly clean up the SocketInterfaceSwap.

Commit Message: Fix flaky test DeferredLoggingWithRetransmission
Risk Level: N/A, just testing

### CODE REVIEW

File: test/integration/quic_http_integration_test.cc [hunk 1]
```diff
@@ -1392,16 +1392,16 @@ TEST_P(QuicHttpIntegrationTest, DeferredLoggingWithRetransmission) {
...
-  socket_swap.write_matcher_->setWriteOverride(nullptr);
+  {
+    SocketInterfaceSwap socket_swap(downstreamProtocol() == Http::CodecType::HTTP3
+                                        ? Network::Socket::Type::Datagram
+                                        : Network::Socket::Type::Stream);
+    Network::IoSocketError* ebadf = Network::IoSocketError::getIoSocketEbadfInstance();
+    socket_swap.write_matcher_->setDestinationPort(lookupPort("http"));
+    socket_swap.write_matcher_->setWriteOverride(ebadf);
+    upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+    absl::SleepFor(absl::Seconds(TSAN_TIMEOUT_FACTOR));
```
[the remark this answers is not in the document]
Comment (pksohn) [reply test/integration/quic_http_integration_test.cc:1403]: The sleep is to wait for a retransmission to occur (such that [QuicStatsGatherer::OnPacketRetransmitted](https://github.com/envoyproxy/envoy/blob/548d4771d8737fa18b568252d0e3ff833ed3bf64/source/common/quic/quic_stats_gatherer.h#L23) is executed). I couldn't think of a good integration-test-level event to wait on. Is there a good way to either 1) detect a retransmitted packet from the server to downstream, or 2) create a custom event to wait on from the QuicStatsGatherer?

[the remark this answers is not in the document]
Comment (alyssawilk) [reply test/integration/quic_http_integration_test.cc:1403]: can we use timeSystem().advanceTimeWait  ?  I don't remember if this test uses simtime.  If so let's switch over.  if not, maybe we can try to dig up some stats

[the remark this answers is not in the document]
Comment (pksohn) [reply test/integration/quic_http_integration_test.cc]: Thanks, timeSystem().advanceTimeWait seems to work.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**